.gitlab/scripts.yml

```diff
@@ -0,0 +1,203 @@
+        -GNinja
+        -DCMAKE_C_COMPILER=${C_COMPILER} -DCMAKE_CXX_COMPILER=${CXX_COMPILER}
+        -DCMAKE_CUDA_COMPILER=${CUDA_COMPILER} -DCMAKE_BUILD_TYPE=${BUILD_TYPE}
+        -DCMAKE_CXX_FLAGS="${CXX_FLAGS}" -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
+        ${EXTRA_CMAKE_FLAGS} ${CUDA_ARCH_STR} ${CUDA_HOST_STR}
+        -DGINKGO_DEVEL_TOOLS=OFF -DGINKGO_BUILD_REFERENCE=${BUILD_REFERENCE}
+        -DGINKGO_BUILD_OMP=${BUILD_OMP} -DGINKGO_BUILD_CUDA=${BUILD_CUDA}
+        -DGINKGO_BUILD_HIP=${BUILD_HIP}
+        -DGINKGO_BUILD_MPI=${BUILD_MPI} -DGINKGO_MPI_EXEC_SUFFIX=${MPI_SUFFIX}
+        -DMPI_RUN_AS_ROOT=${MPI_AS_ROOT}
+        -DGINKGO_BUILD_HWLOC=${BUILD_HWLOC}
+        -DGINKGO_BUILD_TESTS=ON -DGINKGO_BUILD_EXAMPLES=ON
+        -DGINKGO_FAST_TESTS=${FAST_TESTS}
+        -DGINKGO_MIXED_PRECISION=${MIXED_PRECISION}
+        -DGINKGO_CONFIG_LOG_DETAILED=${CONFIG_LOG}
+        -DGINKGO_DPCPP_SINGLE_MODE=${DPCPP_SINGLE_MODE}
+        -DGINKGO_RUN_EXAMPLES=${RUN_EXAMPLES}
+        -DGINKGO_EXPORT_BUILD_DIR=${EXPORT_BUILD_DIR}
+    - ninja -j${NUM_CORES} -l${CI_LOAD_LIMIT} install
+    - |
+        (( $(ctest -N | tail -1 | sed 's/Total Tests: //') != 0 )) || exit 1
+    - ctest -V
+    - ninja test_install
+    - pushd test/test_install
+    - ninja install
+    - popd
+    - |
+      if [ "${RUN_EXAMPLES}" == "ON" ]; then
+         export EX_ARG="reference"
+         ninja run_all_examples
+         ninja validate_all_examples
+         if [ "{BUILD_OMP}" == "ON" ]; then
+            export EX_ARG="omp"
+            ninja run_all_examples
+            ninja validate_all_examples
+        fi
+        if [ "{BUILD_CUDA}" == "ON" ]; then
+            export EX_ARG="cuda"
+            ninja run_all_examples
+            ninja validate_all_examples
+        fi
+        if [ "{BUILD_HIP}" == "ON" ]; then
+            export EX_ARG="hip"
+            ninja run_all_examples
+            ninja validate_all_examples
+        fi
+      fi
+    - if [ -n "${SYCL_DEVICE_TYPE}" ]; then unset SYCL_DEVICE_TYPE; fi
+    - if [ -n "${SYCL_DEVICE_FILTER}" ]; then unset SYCL_DEVICE_FILTER; fi
+    - if [ "${EXPORT_BUILD_DIR}" == "ON" ]; then ninja test_exportbuild; fi
+  dependencies: []
+
+
+.horeka_test_template: &horeka_test
+  stage: test
+  before_script:
+    # Sanity checks
+    - if [ -z "${USE_NAME}" ]; then exit 111; fi
+    - if [ -z "${SLURM_PARTITION}" ]; then exit 222; fi
+    - if [[ ! "${SLURM_GRES}" =~ "^gpu*"  ]]; then export NVIDIA_VISIBLE_DEVICES=void; fi
+  variables:
+    GIT_CLEAN_FLAGS: none
+  script:
+    # We should be reusing the previous image and build directory.
+    # Speedup the tests by limiting the number of OMP threads to CI parallelism.
+    - export OMP_NUM_THREADS=${CI_PARALLELISM}
+    # The test job should have the name name as the image job, only changing
+    # build to test
+    - cd ${CI_PROJECT_DIR}${CI_PROJECT_DIR_SUFFIX}
+    - cd ${CI_JOB_NAME/test/build}
```

Comment:
It depends on the point of view. Here, I'm saying that we suppose the `test` job has exactly the same as the `build` one, only that `build` is changed to `test` (so that we can do this simple string replace in the other direction to find the build directory).